pkg/controller/operators/catalog/step_ensurer.go

```diff
@@ -349,3 +350,20 @@ func (o *StepEnsurer) EnsureConfigMap(namespace string, configmap *corev1.Config
 	status = v1alpha1.StepStatusPresent
 	return
 }
+
+func mergedOwnerReferences(in ...[]metav1.OwnerReference) []metav1.OwnerReference {
+	uniques := make(map[metav1.OwnerReference]struct{})
+	for _, refs := range in {
+		for _, ref := range refs {
+			uniques[ref] = struct{}{}
```

Comment:
The only potential concern I have is whether we need to check if that `ref` key already exists, but I'm almost positive we're safe when we use this implementation?

Reply:
Yeah, this looks like it would be safe re-reading this again as we're only initializing that uniques map.